src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -938,13 +951,13 @@ int runShellCmd(const std::string& cmd, std::string& output)
         ret = -EINVAL;
     }
 
-    close(stderr_fds[0]);
     dup2(stderr_fds[1], 2);
     std::shared_ptr<FILE> pipe(popen(cmd.c_str(), "r"), pclose);
-    close(stderr_fds[1]);
+    close(stderr_fds[1]); 
```

Comment:
extra space in the end.

Reply:
Perhaps it would make sense to use something like a Python-esque https://www.boost.org/doc/libs/1_71_0/doc/html/boost_process/tutorial.html or https://www.boost.org/doc/libs/1_71_0/doc/html/interprocess.html  instead of a low-level C interface from the 70's impossible to debug...